File: golang/main_test.go
package main

import "testing"

func TestIsPalindrome(T *testing.T) {
	palindromes := []int{33533, 222222, 66455466}
	nonPalindromes := []int{665, 244, 188888888, 10}
	for _, input := range palindromes {
		if isPalindrome(input) == false {
			T.Error(input)
		}
	}
	for _, input := range nonPalindromes {
		if isPalindrome(input) == true {
			T.Error(input)
		}
	}
}

func TestGetLargestPalindrome(T *testing.T) {
	//since we are given the palindrome for 2 digits, our generalized approach should work and give
	//the expected answer
	if value := getLargestPalindrome(2); value != 9009 {
		T.Error(value, "Approach failed")

	}
}

File: python/main_test.py
"""find the largest palindrome"""
import main


def test_is_palindrome():
    """tests the palindrome check function"""
    for _input in [33, 565, 77733777]:
        assert main.is_palindrome(_input) is True
    for _input in [334, 56, 7773377]:
        assert main.is_palindrome(_input) is False


def test_get_largest_palindrome():
    """tests the function that gets the largest palindrome"""
    assert main.get_largest_palindrome(2) == 9009
